lib/model/narrow.dart

```diff
@@ -286,4 +286,29 @@ class DmNarrow extends Narrow implements SendableNarrow {
   int get hashCode => Object.hash('DmNarrow', _key);
 }
 
-// TODO other narrow types: starred, mentioned; searches; arbitrary
+class MentionsNarrow extends Narrow {
+  const MentionsNarrow();
+
+  @override
+  bool containsMessage(Message message) {
+    return message.flags.any((flag) =>
+      switch (flag) {
+        MessageFlag.mentioned => true,
+        MessageFlag.wildcardMentioned => true,
+        _ => false
```

Comment:
So in https://github.com/zulip/zulip-flutter/pull/807#discussion_r1693650623 the point was to make sure that when we add a new mentions-related flag, we don't forget to handle it here. This version doesn't really help with that. :slightly_smiling_face:

Instead, list all the flags. That incurs a small cost when we add a non-mentions-related flag, but it's worth it for preventing the class of bug where we forget this for a mentions-related flag.

Reply:
Good thought, thanks!